src/storm-pars/modelchecker/results/RegionCheckResult.cpp

```diff
@@ -109,7 +109,7 @@ namespace storm {
         }
         
         template<typename ValueType>
-        void RegionCheckResult<ValueType>::filter(QualitativeCheckResult const& filter) {
+        void RegionCheckResult<ValueType>::filter([[maybe_unused]] QualitativeCheckResult const& filter) {
```

Comment:
@tquatmann should we throw an error here instead?

Reply:
I *think* the model-handling in CLI always calls the filter method.

IMO the proper fix would be to store the initial state for which this RegionCheckResult applies. Then, the filter method can assert that  the given qualitativeCheckResult is true for exactly that state. We also do this for the [ExplicitParetoCurveCheckResult](https://github.com/moves-rwth/storm/blob/d4a755f5f0a095810d873af18fcb240eb458fef8/src/storm/modelchecker/results/ExplicitParetoCurveCheckResult.cpp#L52)

However, I'm in favour of doing this in a separate pull request, given that this one is already big.